pyresample/geometry.py

```diff
@@ -440,6 +441,19 @@ def __init__(self, lons, lats, nprocs=1):
         elif lons.ndim != 2:
             raise ValueError('2 dimensional lon lat grid expected')
 
+def get_array_hashable(arr):
+    """Compute a hashable form of the array `arr`.
+
+    Works with numpy arrays, dask.array.Array, and xarray.DataArray.
+    """
+
+    if isinstance(arr, DataArray) and np.ndarray != DataArray: # look for precomputed value
+        return arr.attrs.get('hash', get_array_hashable(arr.data))
+    else:
+        try:
+            return hex(hash(arr)) # dask array
+        except TypeError:
+            return arr.view(np.uint8) # np array
```

Comment:
Returning a view of the array as a hash?

Reply:
No, it's returning something hashable from the array